src/net/ipv4.c

```diff
@@ -337,9 +337,12 @@ static int ipv4_tx ( struct io_buffer *iobuf,
 	if ( ( next_hop.s_addr != INADDR_BROADCAST ) &&
 	     ( ( miniroute = ipv4_route ( sin_dest->sin_scope_id,
 					  &next_hop ) ) != NULL ) ) {
-		iphdr->src = miniroute->address;
 		netmask = miniroute->netmask;
 		netdev = miniroute->netdev;
+
+        struct settings *settings = netdev_settings ( netdev );
+        fetch_ipv4_setting ( settings, &ip_setting, &iphdr->src );
```

Comment:
This code is on the transmit datapath, which is latency sensitive and cannot therefore perform operations such as fetching settings.